open-api/rest-catalog-open-api.yaml

```diff
@@ -3647,6 +4080,105 @@ components:
             type: integer
           description: "List of equality field IDs"
 
+    PlanTableScanRequest:
+      type: object
+      properties:
+        snapshot-id:
+          description:
+            Identifier for the snapshot to scan in a point-in-time scan
+          type: integer
+          format: int64
+        select:
+          description: List of selected schema fields
+          type: array
+          items:
+            $ref: '#/components/schemas/FieldName'
+        filter:
+          description:
+            Expression used to filter the table data
+          $ref: '#/components/schemas/Expression'
+        case-sensitive:
+          description: Enables case sensitive field matching for filter and select
+          type: boolean
+          default: true
+        use-snapshot-schema:
+          description:
+            Whether to use the schema at the time the snapshot was written.
+
+            When time travelling, the snapshot schema should be used (true).
```

Comment:
iirc, there are two reasons:
1. We want the client to be explicit about which snapshot they want to read
2. This simplifies the request because there's only one way to read historical data

Reply:
Yeah, clients currently specify the snapshot ID however there needs to be a mechanism for distinguishing which schema gets used based on if it's a time travel by a specific snapshot ID or if it's a time travel by branch. The client has that context, and it's easier for it to determine which schema should be used. The request input is kept simpler by having just a snapshot ID for time travel as @danielcweeks said rather than having a mix of different options.

Reply:
Thanks @danielcweeks and @amogh-jahagirdar for inputs. Sorry I didn't realized there are differences between a time travel by a specific snapshot ID and by branch name, in terms of which schema to use.

To help clarify, could we add a link here for further reading, or provide a detailed explanation? For example: 
```
- When scanning with a snapshot ID, clients should use the snapshot schema (true).
- When scanning with a branch name, clients should use the table schema (false). Note that clients still send the snapshot ID rather than the branch name in the request.
```

Reply:
@flyrain All good! To clarify, the distinction in the schema that gets used during time travel is something that got established in the reference implementation but is not actually defined in the spec itself. 


As for the rationale behind this behavior in the reference implementation please refer to
https://github.com/apache/iceberg/pull/9131/files.

I don't think I'd spec'ing out which ones clients should use in which situation because that's not even in the table spec itself.

I do think it's probably beneficial to provide some more context as to why this field exists, which is to enable clients to align with the reference implementation.

Reply:
Thanks for your insights@amogh-jahagirdar @flyrain 

However I actually think the current description for this is pretty straightforward and do not think we need to explain in the spec the reference implementation context around "snapshot schema" .

cc @rdblue @danielcweeks if you think we should clarify this further or keep as is.

Reply:
It's OK as is, I do think adding that historical context from the reference implementation is valuable for readers because without it the utility of the client side flag is unclear without digging through PRs or discussion threads.

Def not a blocker imo, if others think this context is useful I think we can address this in a follow on.